### PR TITLE
v0.4.0

### LINKED ISSUE #38: Legend in autographs() only prints categories as TRUE/FALSE
The legend when plotting `diff_model` objects using the `autographs()` function shows the categories as 'TRUE' or 'FALSE' rather than 'Infected' or 'Susceptible' etc.

``` r
manynet::autographs(migraph:: play_diffusion(manynet::ison_adolescents, recovery = 0.2), node_color = "Infected") + ggplot2::guides(colour = ggplot2::guide_legend(""))
#> Registered S3 method overwritten by 'manynet':
#>   method          from     
#>   print.tbl_graph tidygraph
```

<sup>Created on 2023-12-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### LINKED ISSUE #43: Color scales not working for edge colors in `autographr()`
At different stages of how we code the autograph family of functions, we map variable to aesthetics or not to change plot aesthetics. This makes so that sometimes scales work as expected and other times not. This issue might require we refactor the code to make it more concise and consistent when it comes to mapping variable to aesthetics. 

`autographr(ison_physicians[[1]], edge_color = "type") + scale_color_sdgs()` is equal to `autographr(ison_physicians[[1]], edge_color = "type") + scale_color_rug()` and to `autographr(ison_physicians[[1]], edge_color = "type")`

### LINKED ISSUE #45: Make `autographs()` plot a limited number plots when lists have many elements
When lists have to many elements, `autographs()` should illustrate a network at the start and end of a set of changes to avoid unnecessary warnings and plot and label overlaps. This might require we add an extra argument to `autographs()`.

### LINKED ISSUE #47: Add Harry Potter data
Add data for all Harry Potter books here (https://www.stats.ox.ac.uk/~snijders/siena/HarryPotterData.html).

### LINKED ISSUE #48: Add another argument here that allows users to specify whether the layout from the first, last, or average/mix of the two is used for both waves.

### LINKED ISSUE #49: Add Sampson network data for testing and illustration purposes
Look at `networkdata::sampson()` for reference. The Sampson network data a small longitudinal, multiplex, signed well-known dataset.

### LINKED ISSUE #52: Add legends by default to `autographr()` for multiple colors

### PR DESCRIPTION
**Pull request by jhollway**
# Description

2023-12-24

## Package

- Major overhaul of documentation
  - Moved function description up so that it appears in tooltips
  - Split documentation of many functions to provide more space for explanation
  - Moved some functions around for more thematic documentation
- Added `pkg_data()` to report an overview of data contained within the package(s)
- Updated README and DESCRIPTION with new organisational affiliation: stocnet/manynet
- Updated tests to be consistent with future `{ggplot2}` release

## Making

- Migrated `play_diffusion()` and `play_diffusions()` from `{migraph}`
  - Including class creation, and `print()`, `summary()`, and `plot()` methods
- Migrated `play_learning()` and `play_segregation()` from `{migraph}`
  - Including class creation, and `print()`, `summary()`, and `plot()` methods
- Fixed bug in `create_tree()` where it was not returning a two-mode network correctly

## Modifying

- Added `as_diffusion()` to coerce a table of diffusion events into diff_model class
  - `as_*()` functions are now considered modifications
- Completed a full range of tidy verbs for nodes
  - Added `mutate_nodes()`
  - Added `filter()` and `filter_nodes()`
  - Added `rename_nodes()`
- Completed a full range of tidy and igraph verbs for ties
  - Added `bind_ties()`
  - Added `delete_ties()`
- Added `to_tree()` to find one or more spanning trees amongst a network's ties
- Added `from_ties()` to collect multiple networks into a multiplex network

## Marking

- Renamed `is.igraph()` to `is_igraph()` for igraph v2.0.0
- Added `is_list()` for identifying a list of networks
- Migrated `node_is_core()`, `node_is_cutpoint()`,  `node_is_fold()`, `node_is_isolate()`,  `node_is_mentor()` from `{migraph}`
- Migrated `node_is_exposed()`, `node_is_infected()`, `node_is_latent()`, `node_is_recovered()` from `{migraph}`
- Migrated `node_is_max()`, `node_is_min()`, `node_is_random()` from `{migraph}`
- Migrated `tie_is_bridge()`, `tie_is_loop()`,  `tie_is_multiple()`, `tie_is_reciprocated()` from `{migraph}`
- Added `tie_is_feedback()`
- Migrated `tie_is_max()`, `tie_is_min()` from `{migraph}`
- Added `tie_is_random()`

## Mapping

- Added `scale_edge_color_centres()`,  `scale_edge_color_ethz()`,  `scale_edge_color_iheid()`,  `scale_edge_color_rug()`,  `scale_edge_color_sdgs()`
  - Fixed bug that meant edge scales were unavailable (closes #43)
- `autographr()` now provides legends by default where multiple colours are used (closes #52)
- `autographs()` now labels legends correctly for binary variables (closes #38)
- `autographs()` now graphs just the first and last networks in a list (closes #45)
- `autographs()` now includes an option whether the layout should be based on the first, last, or both of two networks (closes #48)

## Data

- Renamed `ison_konigsberg` to `ison_koenigsberg` and named the bridges
- `ison_algebra` now in long multiplex format
- `ison_karateka` now weighted, anonymous members are named by number, and "obc" variable renamed "allegiance"
- `ison_lawfirm` enlarged from 36 to 71 nodes and now consists of three multiplex, directed networks
- `ison_southern_women` names are now title case
- Added `ison_hightech`, a multiplex, directed network from Krackhardt 1987
- Added four `ison_monastery` datasets, three of which are signed and weighted, and the other is longitudinal, from Sampson 1969 (closes #49)
- Added six `ison_potter` datasets in a list of networks, from Bossaert and Meidert 2013 (closes #47)
- Added `ison_usstates` data on the contiguity of US states, from Meghanathan 2017

# Checklist:

- PR form
  - [x] Description above itemizes changes under subtitles, e.g. "## Data""
  - [x] Any closed, fixed, or related issues are referenced and explained in the description above, e.g. "Fixed #0 by adding A"
  - [x] Package builds on my OS without issues
- PR checks all pass for latest commit
  - [ ] CodeFactor check: Package improves or maintains good style
  - [x] Package builds on Mac
  - [x] Package builds on Windows
  - [x] Package builds on Linux
  - [ ] CodeCov check: Package improves or maintains good test coverage
- Documentation
  - [x] Any new or modified functions or data have roxygen style documentation in their .R scripts
  - [x] Longer functions are commented inline or broken down into helper functions so that it is easier to debug in the future
  - [x] PR description above and the NEWS.md file are aligned
  - [x] DESCRIPTION file version is bumped by the appropriate increment (major, minor, patch)

### CODE REVIEW

File: R/manip_nodes.R [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+#' @export
+mutate_nodes <- tidygraph::mutate
+
+#' @rdname add_nodes
+#' @importFrom tidygraph mutate
+#' @export
+mutate <- tidygraph::mutate
+
+#' @export
+mutate.igraph <- function(.data, ...){
```
Comment (henriquesposito) [inline R/manip_nodes.R:102]: This needs to be documented consistent as a method, otherwise a warning will be returned...

**Review by henriquesposito — COMMENTED** [review]
Thank you very much for all the work with tidying up the documentation in the package @jhollway , I think it is much better and more consistent. I left a few comments, happy to adress them myself if you would like me to. Thank you.

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,71 @@
+# manynet 1.0.0
+
+2023-12-24
+
+## Package
+
+- Major overhaul of documentation
```
[the remark this answers is not in the document]
Comment (jhollway) [reply NEWS.md:7]: Still major...

File: pkgdown/_pkgdown.yml [hunk 3]
```diff
@@ -32,18 +32,18 @@ navbar:
 reference:
```
[the remark this answers is not in the document]
Comment (jhollway) [reply pkgdown/_pkgdown.yml:32]: No, we would need to serve them with a server/instance running R. The site is currently static.

**Pull request merged**